parser-typechecker/src/Unison/Syntax/FileParser.hs

```diff
@@ -42,14 +41,21 @@ file = do
   -- which are parsed and applied to the type decls and term stanzas
   (namesStart, imports) <- TermParser.imports <* optional semi
   (dataDecls, effectDecls, parsedAccessors) <- declarations
-  env <- case environmentFor (NamesWithHistory.currentNames namesStart) dataDecls effectDecls of
+  env <- case UFN.environmentFor (NamesWithHistory.currentNames namesStart) dataDecls effectDecls of
     Right (Right env) -> pure env
     Right (Left es) -> P.customFailure $ TypeDeclarationErrors es
     Left es -> resolutionFailures (toList es)
+  let accessors :: [[(v, Ann, Term v Ann)]]
+      accessors =
+          [ DD.generateRecordAccessors (toPair <$> fields) (L.payload typ) r
+            | (typ, fields) <- parsedAccessors,
+              Just (r, _) <- [Map.lookup (L.payload typ) (UF.datas env)]
+          ]
+      toPair (tok, typ) = (L.payload tok, ann tok <> ann typ)
```

Comment:
This was moved up from below unchanged so that we'll have the names of all record field accessors and can have those names participate in suffix-based resolution.